spackbot/handlers/reviewers.py

```diff
@@ -118,8 +95,60 @@ async def find_maintainers(gh, packages, repository, pull_request, number):
     return with_maintainers, without_maintainers, all_maintainers
 
 
+async def add_issue_maintainers(event, gh, package_list):
+    """
+    Assign maintainers of packages based on issue title.
+    """
+    # Add extra space to end of title so we catch matches at end
+    title = event.data["issue"]["title"].lower() + " "
+
+    # Does the title have a known package (must have space before and after)
```

Comment:
I would suggest changing this regex strategy a bit.

1. I would use word breaks instead of spaces so that we can catch titles like:
    ```
    python: build fails for some reason
    ```
    We'll have to see if "word characters" includes dashes or not, but I'm hoping you'll be able to use `\b` here.

2. We may want to consider using fuzzy matching so that things like `PyTorch` match `py-torch`. There have been requests for this on Spack's command-line too. This could be saved for a follow-up PR though.

3. Also, I worry that Spack has so many packages that generic words in issue titles may be matched by accident. For example, `benchmark`, `arm` (and other arches), `gcc` (and other compilers), `bash` (and other shells), `python` (and other languages), `cmake` (and other build systems), and `git` (and other VCS) are all words that commonly appear in issue titles even if the issue has nothing to do with that package. I'm a bit concerned about false positives for these kind of titles. Of course, this fear could be completely unfounded. We could deploy it as is and see how many false positives we get. Or we could require a specific format like:
    ```
    installation issue: <package> .*
    <package> .*
    ```

Reply:
That's a good point - I was worried about this too. My thinking is that this is a bit of an experiment in itself, and that we should deploy and then start to keep track of false positives and false negatives, and then adjust accordingly.

Reply:
The only problem is if one or two packages get hit with a lot of false positives and those maintainers get angry at us 😛 

Possible experiment: we could query existing open/closed issue titles and see how many false positives we get. Although it isn't always easy to tell if something is a false positive or not without manually going through all 10K+ titles...

Reply:
Oh that sounds fun! yeah I'll totally do that! I have a meeting in 10 mins but will do after.

Reply:
okay @adamjstewart here is a first look! I could only run this on about 6K issues because I used up my API limit to get them, lol, but I'll keep updating the data as the limit is restored (I think it's like 5k an hour or something?) Anyway, here is the repo:

https://github.com/vsoch/spacksperiment/tree/main/titles

And here are files of matches to the pattern strategy and misses, respectively:

 - https://github.com/vsoch/spacksperiment/blob/main/titles/matches.json
 - https://github.com/vsoch/spacksperiment/blob/main/titles/misses.json

I think you were spot on about the pattern of a package name followed by a colon - I see this pattern too with dashes so we miss a subset of packages. This could be easily fixed if we replace all special characters with spaces. The other thing I see (which I'm not sure is easy to fix) is the case of a title mentioning multiple packages, or a package in the context of others. E.g., a lot of the mpi ones also mention slurm and vice versa

Reply:
I also think we should skip some packages entirely that are just likely to be false positives because they are English words, e.g., "patch"

Reply:
okay just tweaked the regex a little more and updated the spacksperiment repo with that, we now have about 12K results (maybe that's about half?)